futures-util/src/future/either.rs

```diff
@@ -1,55 +0,0 @@
-use futures_core::{Future, Poll, Stream};
-use futures_core::task;
-
-/// Combines two different futures yielding the same item and error
-/// types into a single type.
-#[derive(Debug)]
-pub enum Either<A, B> {
-    /// First branch of the type
-    A(A),
-    /// Second branch of the type
-    B(B),
-}
-
-impl<T, A, B> Either<(T, A), (T, B)> {
-    /// Splits out the homogeneous type from an either of tuples.
-    ///
-    /// This method is typically useful when combined with the `Future::select2`
-    /// combinator.
-    pub fn split(self) -> (T, Either<A, B>) {
```

Comment:
See #580 for modifications related to this. I don't know if `either` is inclined to add something similar, but if not, the method is useful enough that free-standing functions may still be useful.